cleanlab/dataset.py

```diff
@@ -0,0 +1,150 @@
+        idx, scores = _find_largest_pairs(joint_no_diagonal)
+        cutoff = len(joint) ** 2 - len(joint)  # Omit diagonal elements from end of ranking
+        return idx[:cutoff], scores[:cutoff]
+    else:  # symmetric
+        # Sum the upper and lower triangles and remove the lower triangle and the diagonal
+        # This provides values only in (the upper triangle) of the matrix.
+        sym_joint = np.triu(joint) + np.tril(joint).T
+        sym_joint = sym_joint - np.eye(len(sym_joint)) * sym_joint.diagonal()
+        idx, scores = _find_largest_pairs(sym_joint)
+        cutoff = (len(joint) ** 2 - len(joint)) / 2  # Omit lower triangle and diagonal elements
+        return idx[:cutoff], scores[:cutoff]
+
+
+def get_classes_ranked_by_label_noise(joint, descending=True, rank_by="given_label"):
+    """Returns the ordered class indices and associated scores quantifying how
+    noisy (prone to label issues) each class in the dataset is.
+
+    Score values are unnormalized and may tend to be very small. What matters is their relative
+    ranking across the classes."""
+
+    if rank_by == "given_label":
+        # Ranks the label quality of each class based on the given (potentially error-prone) labels
+        # This score represents how many of the given labels in each class likely have issues.
+        scores = joint.sum(axis=1) - joint.diagonal()
+    elif rank_by == "true_label":
+        # Ranks the label quality of each class based on the estimated true labels.
+        # This score represents how likely each class was actually labeled something else.
+        scores = joint.sum(axis=0) - joint.diagonal()
+    elif rank_by == "total_noise":
+        # p(label=k, true_label) + p(label, true_label=k) - 2 * P(label=k, true_label=k)
+        # The error for the class (both in terms of how often the given label is wrong
+        # and how often other classes should be this label.
+        scores = joint.sum(axis=0) + joint.sum(axis=1) - 2 * joint.diagonal()
+    else:
+        raise ValueError(
+            f"parameter 'rank_by' should be 'given_label', 'true_label', or"
+            "'total_noise', but was {rank_by}"
+        )
+    class_indices, scores = np.argsort(scores), np.sort(scores)
+    if descending:
+        class_indices, scores = class_indices[::-1], scores[::-1]
+    return class_indices, scores
+
+
+def get_classes_ranked_by_label_quality(joint, descending=True):
+    """Returns the ordered class indices and associated scores quantifying the quality of the labels
+     in each class (larger number implies higher quality labels) for the joint of a dataset.
+
+    Score values are unnormalized. What matters is relative ranking across the classes."""
+
+    scores = joint.diagonal()
+    class_indices, scores = np.argsort(scores), np.sort(scores)
+    if descending:
+        class_indices, scores = class_indices[::-1], scores[::-1]
+    return class_indices, scores
+
+
+def dataset_label_quality_score(joint):
```

Comment:
could consider alternative name `overall_label_quality_score` or `global_label_quality_score`

Reply:
again consider allowing many of the same args as `find_similar_classes`

Reply:
Changed to `overall_label_health_score` and added args.